internal/config/mount_config.go

```diff
@@ -14,26 +14,58 @@
 
 package config
 
+const (
+	// Default log rotation config values.
+	defaultMaxFileSizeMB = 512
+	defaultFileCount     = 10
+	defaultCompress      = false
+)
+
 type WriteConfig struct {
 	CreateEmptyFile bool `yaml:"create-empty-file"`
 }
 
 type LogConfig struct {
-	Severity LogSeverity `yaml:"severity"`
-	Format   string      `yaml:"format"`
-	FilePath string      `yaml:"file-path"`
+	Severity        LogSeverity     `yaml:"severity"`
+	Format          string          `yaml:"format"`
+	FilePath        string          `yaml:"file-path"`
+	LogRotateConfig LogRotateConfig `yaml:"log-rotate"`
 }
 
 type MountConfig struct {
 	WriteConfig `yaml:"write"`
 	LogConfig   `yaml:"logging"`
 }
 
+// LogRotateConfig defines the parameters for log rotation. It consists of three
+// configuration options:
+// 1. max-file-size-mb: specifies the maximum size in megabytes that a log file
+// can reach before it is rotated. The default value is 512 megabytes.
+// 2. file-count: determines the maximum number of log files to retain after they
+// have been rotated. The default value is 10.
+// 3. compress: indicates whether the rotated log files should be compressed
```

Comment:
Should we write by what factor (~) log-file size gets compressed after using this flag?
Not here, but we can update in log-rotate doc.